src/abstractdatatable/abstractdatatable.jl

```diff
@@ -801,6 +750,165 @@ function Base.hash(dt::AbstractDataTable)
     return @compat UInt(h)
 end
 
+"""
+    denullify!(dt::AbstractDataTable)
+
+Convert columns with a `Nullable` element type without any null values
+to a non-`Nullable` equivalent array type. The table `dt` is modified in place.
```

Comment:
I've changed these back to using copy as you suggested and added a note for `nullify` and `denullify` that if users want fully alias-free copies, they should use `nullify!(deepcopy(dt))` and `denullify!(deepcopy(dt))`. Hopefully if anyone hits this issue and doesn't know why columns are changed across multiple tables, they'll open the help docstrings and see the note.